neurips2022/track2/train/utility_IL.py

```diff
@@ -81,16 +86,23 @@ def load_data_scratch(dataset_path, save_path):
             for i in range(len(image_names) - 1):
                 image = Image.open(os.path.join(path, scenario) + "/" + image_names[i])
                 obs.append([np.moveaxis(np.asarray(image), -1, 0)])
-                sim_time = image_names[i].split("_Agent")[0]
-                sim_time_next = image_names[i + 1].split("_Agent")[0]
-                last_sim_time = image_names[-1].split("_Agent")[0]
+                match = re.search("(.*)_.*", image_names[i])
+                sim_time = match.group(1)
+                match = re.search("(.*)_.*", image_names[i+1])
+                sim_time_next = match.group(1)
+                match = re.search("(.*)_.*", image_names[-1])
+                last_sim_time = match.group(1)
 
                 try:
-                    current_position = vehicle_data[float(sim_time)]["ego"]["pos"]
-                    next_position = vehicle_data[float(sim_time_next)]["ego"]["pos"]
-                    current_heading = vehicle_data[float(sim_time)]["ego"]["heading"]
-                    next_heading = vehicle_data[float(sim_time_next)]["ego"]["heading"]
-                    goal_location = vehicle_data[float(last_sim_time)]["ego"]["pos"]
+                    current_position = vehicle_data[float(sim_time)].ego_vehicle_state.position
+                    next_position = vehicle_data[float(sim_time_next)].ego_vehicle_state.position
+                    current_heading = vehicle_data[float(sim_time)].ego_vehicle_state.heading
+                    next_heading = vehicle_data[float(sim_time_next)].ego_vehicle_state.heading
+
+                    goal_x = vehicle_data[float(sim_time)].ego_vehicle_state.mission.goal.position.x
+                    goal_y = vehicle_data[float(sim_time)].ego_vehicle_state.mission.goal.position.y
+                    goal_z = vehicle_data[float(sim_time)].ego_vehicle_state.mission.goal.position.z
+                    goal_location = np.array([goal_x, goal_y, goal_z])
```

Comment:
Goal location for a given vehicle in a given scenario is fixed and does not change with time. Hence, we may not need to compute it repeatedly at every time point.